src/Microsoft.AspNetCore.Rewrite/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Rewrite.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+        public static Uri ToUri(this HttpRequest request)
```

Comment:
No, we do not use System.Uri, it makes too many modifications on its own. Remove this and use https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Extensions/UriHelper.cs